staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go

```diff
@@ -139,9 +139,11 @@ type ValidationRule struct {
 	// ref: https://github.com/google/cel-spec
 	// The validator will be scoped to the location of the x-kubernetes-validations extension in the schema
 	// and will use `self` to represent the scoped field name.
-	// If an object property collides with a [RESERVED keyword](https://github.com/google/cel-spec/blob/master/doc/langdef.md#syntax), it will be escaped by prepending a _ prefix.
+	// If an object property collides with a [RESERVED keyword](https://github.com/google/cel-spec/blob/v0.6.0/doc/langdef.md#syntax), it will be escaped by prepending a _ prefix.
 	// To prevent this from causing a subsequent collision, all properties with a _ prefix will always be prefixed by __ (generally, N+1 the existing number of _s).
-	// e.g. rule against field "_pre": {"rule": "__pre > 0"}
+	// You should escape the property by prepending a _ prefix. e.g. rule against field "_pre": {"rule": "__pre > 0"}
+	// We will exclude self and reserved identifiers from the root bound variables without escaping.
```

Comment:
It won't be obvious to developers that "reserved identifiers" refers to a different set than "RESERVED keyword" above. 

The reserved identifiers are `"int", "uint", "double", "bool", "string", "bytes", "list", "map", "null_type", "type"` and are defined in https://github.com/google/cel-spec/blob/master/doc/langdef.md#values.

Reply:
Maybe something like: "If a property name is "self" or collides with a reserved language identifier ..., it is not escaped, but it is excluded from the root bound variables and it can only be accessed via "self.<property name>"...

Reply:
Added more details in API spec.

Reply:
> Maybe something like: "If a property name is "self" or collides with a reserved language identifier ..., it is not escaped, but it is excluded from the root bound variables and it can only be accessed via "self.{property name}"...

I like this... it's way simpler to reason about than _ escaping, in my opinion